Fix FrameContextifier

IonBazan opened the pull request:
This PR fixes #1103 when `$file` is same as absolute path.

ste93cry commented on src/Frame.php:
```diff
@@ -92,7 +92,7 @@ public function __construct(?string $functionName, string $file, int $line, ?str
         $this->line = $line;
         $this->rawFunctionName = $rawFunctionName;
-        $this->absoluteFilePath = $absoluteFilePath;
+        $this->absoluteFilePath = $absoluteFilePath ?? $file;
```
In case the frame is internal then `$file` would be set to `[internal]` because we need a placeholder somehow but reporting that value as absolute path of the file would not be correct

ste93cry reviewed (changes requested):
Rather than making the `abs_path` attribute required, which in the case of internal frames would report an incorrect value, I would rather change the frame builder to set its value regardless of whether the `$file` is equal or not to its stripped value. Technically speaking if `$file` is not an absolute path we would still report an incorrect value and the integration would try to read a non-existent file, but it should not happen if the frame is built from a real backtrace

IonBazan commented:
@ste93cry Thanks for the review. I've applied the changes. Not sure if I should pass `$file` as `$absoluteFilePath` here too:
https://github.com/getsentry/sentry-php/blob/65030a1376b900aee06a4f477bbf320bc8dc9c52/src/StacktraceBuilder.php#L73

IonBazan commented on src/FrameBuilder.php:
```diff
@@ -84,7 +84,7 @@ public function buildFromBacktraceFrame(string $file, int $line, array $backtrac
             $line,
             $rawFunctionName,
-            Frame::INTERNAL_FRAME_FILENAME !== $file && $strippedFilePath !== $file ? $file : null,
+            Frame::INTERNAL_FRAME_FILENAME !== $file ? $file : null,
```
I was also thinking about moving this check to `Frame` constructor.

ste93cry replied:
I prefer to leave the responsibility of deciding whether the `$absoluteFilePath` argument should be filled or not to the builder

ste93cry commented:
>Not sure if I should pass `$file` as `$absoluteFilePath` here too

Strictly speaking yes, but we should also strip the prefix from the path like we do for the rest of the frames. We should probably call the frame builder also for this "fake" frame :thinking:

ste93cry commented on src/Integration/FrameContextifierIntegration.php:
```diff
@@ -78,11 +78,12 @@ public function setupOnce(): void
                 continue;
             }
 
-            $sourceCodeExcerpt = $this->getSourceCodeExcerpt($maxContextLines, $frame->getAbsoluteFilePath(), $frame->getLine());
```
Shouldn't we revert every change in this file now that we always set the `$absoluteFilePath` field except when the frame is internal?

IonBazan commented:
@ste93cry You're right. I've squashed my commits so only the required changes are there. Added necessary assertions to `StacktraceBuilderTest` to make sure it won't happen again.

ste93cry reviewed (approved):
Changes look great, thank you for both spotting and solving this bug :sunglasses: Can you please add a CHANGELOG entry before I merge?

ste93cry commented:
@IonBazan maybe it passed under the radar because I edited the comment after posting it, but would you be so kind to add an entry to the CHANGELOG?

IonBazan commented:
@ste93cry Added the changelog entry - feel free to modify it before release if it does not match the standards.
